lnrpc/invoicerpc/invoice.proto

```diff
@@ -144,43 +226,71 @@ message Invoice {
-    increment this index making it monotonically increasing. Callers to the
-    SubscribeInvoices call can use this to instantly get notified of all
-    settled invoices with an settle_index greater than this one.
-    */
-    uint64 settle_index = 17;
+    The "last update" index of this invoice. last_update_index is a 
+    monotonically increasing value that assigns an order to all invoice updates.
+    It can be used by clients to catch up with missed updates. 
 
-    /// Deprecated, use amt_paid_sat or amt_paid_msat.
-    int64 amt_paid = 18;
-
-    /**
-    The amount that was accepted for this invoice, in satoshis. This will ONLY
-    be set if this invoice has been settled. We provide this field as if the
-    invoice was created with a zero value, then we need to record what amount
-    was ultimately accepted. Additionally, it's possible that the sender paid
-    MORE that was specified in the original invoice. So we'll record that here
-    as well.
+    Whenever a change to any invoice in the system is made, last_update_index
+    is incremented and stored in the updated invoice. 
     */
-    int64 amt_paid_sat = 19;
+    uint64 last_update_index = 17;
 
     /**
-    The amount that was accepted for this invoice, in millisatoshis. This will
-    ONLY be set if this invoice has been settled. We provide this field as if
+    The amount that was accepted or settled for this invoice, in millisatoshis.
+    This will be set in both the ACCEPTED and SETTLED invoice states. 
+    
+    In the OPEN state, there may be htlc(s) accepted, but their value is not
+    reported (yet). Only when the sum reaches the invoice amount, the invoice
+    moves to ACCEPTED or SETTLED and amt_paid_msat is reported.
+    
+    We provide this field as if
     the invoice was created with a zero value, then we need to record what
     amount was ultimately accepted. Additionally, it's possible that the sender
     paid MORE that was specified in the original invoice. So we'll record that
     here as well.
     */
     int64 amt_paid_msat = 20;
+
+    enum InvoiceState {
+        /**
+        Invoice are created in the open state. We have not yet received an htlc
+        or the total value of all received htlcs is less than the invoice
+        amount. No htlc has been settled yet. If the total value of received
+        htlcs reaches the invoice amount, the invoice either moves to ACCEPTED
+        or SETTLED based on the hold invoice field value.
+        */
+        OPEN = 0;
+
+        /**
+        In the accepted state, we have received one or more htlcs that add up 
+        to at least the invoice amount. The htlcs can be accepted by calling
+        the SettleInvoice rpc. When an htlc times out, it may be that the
+        invoice moves back to the open state.
+        */
+        ACCEPTED = 1;
+
+        /**
+        In the settled state, the invoice isn't payable anymore. We have
```

Comment:
Non-payable settled invoices is a change from the current implementation.

Reply:
Shouldn't we then also accept (not yet settle) payments with a value less than the invoice amt?

Reply:
Okay, that is for the `OPEN` state. What shall we do in the `ACCEPT` and `SETTLED` states? Do we go into catch all mode?